### PR TITLE
Fix submission status on 'list submissions' page

### PR DESCRIPTION
**Pull request by samuelhwilliams**
## 🎫 Ticket
BAU

## 📝 Description
This was showing the 'status' field of the Submission DB model, rather than the generated status from the submission helper.

We also remove the 'status' column from the Submission table, since this isn't actually used everywhere. For now, we dynamically determine the status based on submission events and form data.

## Screenshots

| Before | After |
|---|---|
| image | image |

## 🧪 Testing
Check the 'list/view submissions' pages in the reports tab.

### Review Readiness
- [x] PR title and description provide sufficient context for reviewers
- [x] Commits are logical, self-contained, and have clear descriptions

### Testing
- [x] I have tested this change and it meets the acceptance criteria for the ticket
- [x] I need the reviewer(s) to pull and run this change locally
- [x] New (non-developer) functionality has appropriate unit and integration tests
- [-] End-to-end tests have been updated (if applicable)
- [-] Edge cases and error conditions are tested

### CODE REVIEW

File: tests/integration/deliver_grant_funding/routes/test_reports.py [hunk 1]
```diff
@@ -796,7 +796,11 @@ def test_no_submissions(self, authenticated_grant_member_client, factories, db_s
         assert "No submissions found for this monitoring report" in response.text
 
     def test_based_on_submission_mode(self, authenticated_grant_member_client, factories, db_session):
-        report = factories.collection.create(grant=authenticated_grant_member_client.grant, name="Test Report")
+        report = factories.collection.create(
+            grant=authenticated_grant_member_client.grant,
+            name="Test Report",
+            create_completed_submissions_each_question_type__test=1,
```
Comment (MarcUsher) [inline tests/integration/deliver_grant_funding/routes/test_reports.py:802]: 🤏  lil nit/for us to think about (or just my Monday brain) - this helper method on the collection factory maybe needs renaming? It's not creating 'Completed' submissions in the status sense, rather just submissions that have all the questions filled out, maybe just a little misleading to have 'completed' in the name when we use that to mean a submission that's been completed filled in, all sections marked as complete and the 'submit' button clicked.

**Review by MarcUsher — APPROVED** [review]
Nice catch, this seems sensible! Pulled locally and works for me (didn't clock before that the submission status wasn't updating but it is working nicely now). 

Just one thought below but it's not a crucial so approving 👍
